witchslabyrinth/WitchsLabyrinthGame
# New Feature: Actor swapping

# Actor Swapping
- Press 1 to switch to Oliver
- Press 2 to switch to Cat
- Use arrow keys to switch camera perspective for currently-controlled actor

### Implementation Notes
- Required big refactor in camera/perspective functionality
    - refactored update routines in CameraController.cs
    - created Perspective.cs class to hold data (camera position/angle, player movement type, orthographic/not, etc) for each camera perspective
    - moved perspective-swapping logic to PerspectiveController.cs
        - removed GameManager.cs because there wasn't any code left in it after this lol


## Things to Test:
- Test that switching between Oliver/Cat works when you'd expect it to, and doesn't when it shouldn't (i.e. interacting with Zodiac Puzzle)
    - When you're controlling Oliver in 3D perspective and you switch to Cat (who was previously in 2D perspective), should the camera switch to Cat's 2D perspective, or stay in Oliver's 3D perspective?
    - i.e. make sure swapping back/forth feels natural and intuitive
- Test that each character interacts with puzzles as you'd expect them to
    - Should the cat be able to solve the Zodiac puzzle? (it currently can)
- Try and mess with the camera perspectives on Oliver/Cat and see if anything breaks/feels weird
- Make sure the game works as it did before
- Try and break my shit lol

## Known Bugs:
- When you move the 3D camera around for Oliver, it moves the cat's camera as well (and visa versa)
- Camera is deactivated in dialogue with Liar's puzzle (may be a problem in `develop` as well)
